File: easy/cpp/0038.cpp
class Solution {
public:
    string countAndSay(int n) {
        string start = "1";
        int epoch = 1;
        while(epoch < n){
            string new_start = "";
            char prev = start[0];
            int size = start.length();
            int cnt = 1;
            for(int i = 1; i < size; i++){
                if(prev != start[i]){
                    new_start += to_string(cnt) + prev;
                    cnt = 0;
                    prev = start[i];
                }
                cnt++;
            }
            new_start += to_string(cnt) + prev;
            epoch++;
            start = new_start;
        }
        return start;
    }
};

File: easy/python/0038.py
class Solution:
    def countAndSay(self, n: 'int') -> 'str':
        start = "1"
        epoch = 1
        while epoch < n:
            new_start = ""
            cnt = 1
            prev = start[0]
            for s in start[1:]:
                if s == prev:
                    cnt += 1
                else:
                    new_start += str(cnt) + prev
                    cnt = 1
                    prev = s
            new_start += str(cnt) + prev
            epoch += 1
            start = new_start
        return start
        